Modules/_pickle.c

```diff
@@ -6229,6 +6277,39 @@ load_build(UnpicklerObject *self)
         Py_INCREF(slotstate);
         Py_DECREF(tmp);
     }
+    /* proto 5 addition: state can embed a callable state setter */
```

Comment:
For now it is not you are right. I do not really know whether we should enable this change only for protocol 5. Enabling this for lower protocol will generate cryptic errors when trying to read pickle string of recent pythons from older pythons (although doing this is not recommended anyways). 

Another example of a non-opcode-related protocol addition (not written by me)
https://github.com/python/cpython/blob/f5f336a819a3d881bb217bf8f9b5cacba03a4e45/Modules/_pickle.c#L6220-L6222

Reply:
If pierreglaser#2 is merged (which I think it should), then this `state_setter` PR is really independent of the protocol version since a pickle generated using that new feature can be loaded by any older version of Python `pickle.load`.